### PR TITLE
Issue 9803 ng select component enhancement

### PR DESCRIPTION
**Pull request by oswaldogallango**

### CODE REVIEW

File: src/view/components/common/login-as/login-as.ts [hunk 1]
```diff
@@ -54,4 +56,44 @@ export class LoginAsComponent extends BaseComponent {
...
+        this.filteredLoginAsUsersResults = [];
+        for(let i = 0; i < this.userLists.length; i++) {
+            let user = this.userLists[i];
+            if(user.fullName.toLowerCase().indexOf(event.query.toLowerCase()) >= 0) {
+                this.filteredLoginAsUsersResults.push({
+                    label: user.fullName,
+                    value: user.userId,
+                });
+            }
+        }
```
Comment (freddyucv) [inline src/view/components/common/login-as/login-as.ts]: better use:

this.filteredLoginAsUsersResults = this.userLists.
          filter( user => user.fullName.toLowerCase().indexOf(event.query.toLowerCase()) >= 0)
          .map( user=> {
                     label: user.fullName,
                     value: user.userId,
                 });

File: src/view/components/common/login-as/login-as.ts [hunk 2]
```diff
@@ -54,4 +56,44 @@ export class LoginAsComponent extends BaseComponent {
...
+            for(let i = 0; i < this.userLists.length; i++) {
+                let user = this.userLists[i];
+                this.filteredLoginAsUsersResults.push({
+                    label: user.fullName,
+                    value: user.userId,
+                });
+            }
+        }, 100);
+    }
+
```
Comment (freddyucv) [inline src/view/components/common/login-as/login-as.ts]: why we need the timeout?, we should look for another way to do this with the timeout

File: src/view/components/site-selector/dot-site-selector-component.ts [hunk 3]
```diff
@@ -38,4 +42,44 @@ export class SiteSelectorComponent extends BaseComponent {
...
+        setTimeout(() => {
+            for(let i = 0; i < this.sites.length; i++) {
+                let site = this.sites[i];
+                this.filteredSitesResults.push({
+                    label: site.hostName,
+                    value: site.identifier,
+                });
+            }
+        }, 100);
+    }
```
Comment (freddyucv) [inline src/view/components/site-selector/dot-site-selector-component.ts]: why we need the timeout?, we should look for another way to do this with the timeout

File: src/view/components/site-selector/dot-site-selector-component.ts [hunk 4]
```diff
@@ -38,4 +42,44 @@ export class SiteSelectorComponent extends BaseComponent {
...
+        this.filteredSitesResults = [];
+        for(let i = 0; i < this.sites.length; i++) {
+            let site = this.sites[i];
+            if(site.hostName.toLowerCase().indexOf(event.query.toLowerCase()) >= 0) {
+                this.filteredSitesResults.push({
+                    label: site.hostName,
+                    value: site.identifier,
+                });
+            }
+        }
```
Comment (freddyucv) [inline src/view/components/site-selector/dot-site-selector-component.ts]: better use:

this.filteredSitesResults = this.sites
      .filter( site => site.hostName.toLowerCase().indexOf(event.query.toLowerCase()) >= 0)
      .map( site => {
                     label: site.hostName,
                     value: site.identifier,
                 } );

File: src/view/components/common/login-as/login-as.ts (hunk 2 above)
Comment (oswaldogallango) [inline src/view/components/common/login-as/login-as.ts]: Yes, the timeout seems to to be required http://forum.primefaces.org/viewtopic.php?f=35&t=47931&p=148519&hilit=autocomplete#p148519

File: src/view/components/site-selector/dot-site-selector-component.ts (hunk 3 above)
Comment (oswaldogallango) [inline src/view/components/site-selector/dot-site-selector-component.ts]: Yes, the timeout seems to to be required http://forum.primefaces.org/viewtopic.php?f=35&t=47931&p=148519&hilit=autocomplete#p148519

File: src/view/components/site-selector/dot-site-selector-component.ts (hunk 4 above)
Comment (oswaldogallango) [inline src/view/components/site-selector/dot-site-selector-component.ts]: done

File: src/view/components/common/login-as/login-as.ts (hunk 1 above)
Comment (oswaldogallango) [inline src/view/components/common/login-as/login-as.ts]: done

**Pull request merged**